onnxruntime/contrib_ops/cpu/math/sparse_dense_matmul.cc

```diff
@@ -84,13 +84,37 @@ struct SparseToDenseCsr {
     const auto& b_dims = B.Shape().GetDims();
     const auto& out_dims = output.Shape().GetDims();
     auto csr_view = A.AsCsr();
-
-    ConstSparseMatrixMap<T> map_A(a_dims[0], a_dims[1], A.NumValues(),
-                                  csr_view.Outer().Data<int64_t>(),
-                                  csr_view.Inner().Data<int64_t>(),
+    const Eigen::Index* inner_index_pointer = nullptr;
+    const Eigen::Index* outer_index_pointer = nullptr;
+    std::unique_ptr<Eigen::Index[]> buffer_holder_inner, buffer_holder_outer;
+    if constexpr (std::is_same<Eigen::Index, int64_t>::value) {
+      // On macOS the following reinterpret_cast is necessary because Eigen::Index is an alias of `long` but int64_t is `long long`. Though they have the same size, compilers still do not allow an implicit casting between them.
+      inner_index_pointer = reinterpret_cast<const Eigen::Index*>(csr_view.Inner().Data<int64_t>());
+      outer_index_pointer = reinterpret_cast<const Eigen::Index*>(csr_view.Outer().Data<int64_t>());
+    } else {
+      const size_t inner_size = narrow<size_t>(csr_view.Inner().Shape().Size());
+      const size_t outer_size = narrow<size_t>(csr_view.Outer().Shape().Size());
+      buffer_holder_inner.reset(new Eigen::Index[inner_size]);
+      buffer_holder_outer.reset(new Eigen::Index[outer_size]);
+      inner_index_pointer = buffer_holder_inner.get();
+      outer_index_pointer = buffer_holder_outer.get();
+#ifdef _WIN32
+#pragma warning(push)
+// The copy below may loss precision. Ignore the warnings for now
+#pragma warning(disable : 4244)
+#endif
+      std::copy_n<const int64_t*, size_t, Eigen::Index*>(csr_view.Inner().Data<int64_t>(), inner_size, buffer_holder_inner.get());
```

Comment:
Since we do not care about performance if this code (much), I would do `std::transform` with `narrow`

Reply:
I will update it.